### DISCUSSION TITLE
Limit size for circular-progress

### DISCUSSION THREAD

**Discussion by patrikha** [elkowar/eww#713] [Q&A]
Hi, is it possible to limit the size of the circular-progress widget not to fill its container. I would like to be able to set the margins but that don't seem to work.

Any suggestions?

**Comment by patrikha** [comment]
Ok looked at the code an realized I could control the size by changing the circular-progress child object.

**Reply by barddust** [reply]
can you explain that more precisely. I am encountering that problem, can not change the size of circular-progress.

**Reply by patrikha** [reply]
If I remember correctly the size of the circular-progress is determined by its content so in my case:
```
(defwidget cpu [average_percent]
  (box
    :class "cpu_module"
    :vexpand "false"
    :hexpand "false"
    (circular-progress
      :value average_percent
      :class "circularbar"
      :thickness 2
      (button
        :onclick "${EWW_CMD} open --toggle cpu_ctl"
        :class "circulartext"
        :tooltip "cpu average usage ${average_percent}%"
        "CPU"))))
```
the button size with the CPU label.

**Reply by barddust** [reply]
The key is `:thickness` ? I've tried that and it does not change. I am just finding another way out...

And would you mind telling me what `EWW_CMD` exactly is? All I know is `${EWW_CMD} update` to update the value of variables.

**Reply by patrikha** [reply]
EWW_CMD is a [magic variable](https://elkowar.github.io/eww/magic-vars.html) and in this case used to toggle (open/close) a CPU graph widget.
The css style of the button will set the size of the surrounding circular-progress
